vissl/utils/io.py

```diff
@@ -51,7 +51,7 @@ def create_file_symlink(file1, file2):
         logging.info(f"Could NOT create symlink. Error: {e}")
 
 
-def save_file(data, filename):
+def save_file(data, filename,rewrite=True):
```

Comment:
So since we do more overwriting vs. appending, instead of rewrite=True, I would use `append=False`.

1. Add option append, with default False to #save_file.
2. Add in code to support json append=True and append=False.
3. Change [here](https://github.com/facebookresearch/vissl/blob/4b19ab28953470d615a25273fa22eab659528867/vissl/hooks/log_hooks.py#L498) to use append=True. This is the only use of #save_file in the codebase using json format. Global search save_file to validate this.

CC: @prigoyal What do you think?

Reply:
I like the idea of `append` instead of `rewrite`. All your recommendations make sense. A few additional thoughts:
1. let's update the docstring of this function to mention clearly the `append` is applicable only for the `json` format.
2. for the above reason, we might want to pick a more verbose name `append_to_json` for instance.

Reply:
Thanks for feedback @iseessel and @prigoyal ! I have addressed your inputs in e9efe76.

- Switched the variable to ```append_to_json```  and set the default value to false.
- Update the doc-string of the save_file function to inform the user about the same.

Do let me know if further changes are to be made to polish the PR!